pkg/operator/operator.go

```diff
@@ -111,13 +111,26 @@ func (c *vSphereProblemDetectorController) sync(ctx context.Context, syncCtx fac
 		return err
 	}
 
+	availableCnd := operatorapi.OperatorCondition{
+		Type:   controllerName + operatorapi.OperatorStatusTypeAvailable,
+		Status: operatorapi.ConditionTrue,
+	}
+
 	// TODO: Run in a separate goroutine? We may not want to run time-consuming checks here.
 	if platformSupported && time.Now().After(c.nextCheck) {
 		delay, err := c.runChecks(ctx)
 		if err != nil {
-			// This sets VSphereProblemDetectorControllerDegraded condition
-			return err
+			// Do not return the error, it would degrade the whole cluster.
```

Comment:
Can we report errors from talking to openshift's api-server differently from error from talking to vcenter api server?

Reply:
How would we use the information? Both APi server and vCenter errors will result in an alert and IMO a generic one is enough.

Reply:
We would use that information by putting specific error in alert. The approach here seems backward and does not help us with - https://bugzilla.redhat.com/show_bug.cgi?id=1929780

Reply:
The information we can put in alert will be very generic in both cases, because we don't have detailed error message in a metric. In addition, `runChecks()` can fail for variety of reason on OCP side, for example a secret is missing or has invalid format. I don't want to have separate alerts for them, it would be too many, and at the same time you can't unify them into a generic "Failed to connect to API server", because it's not true.

I want to have alert something like *"Failed to initialize vsphere-problem-detector periodic checks for X minutes. Please see ClusterOperator "storage" conditions for detailed error message, "oc get clusteroperator storage -o yaml", and look for message in Available condition"* .

The bug you linked is to have specific alert for every check, not to have alert for every error of every check.

Reply:
My main concern is - since we are using kubeClient directly for fetching list of nodes - we could have an sporadic failure in talking to api-server, which would result in a recorded metric and hence an alert (eventually).

Reply:
Moved away from API calls and using informers for everything. Now either there is something wrong with a secret (can't be parsed, does not exist, ...) or vCenter returns a bad response.